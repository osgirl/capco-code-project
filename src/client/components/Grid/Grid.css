html,
body,
.section-grid {
  height: 100%;
  margin: 2px;
}

.section-grid {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows:  (auto-fill); 
  grid-template-areas: "cell--large cell--large";
}

.img-content {
  max-width: 100%;
  padding: 1%;
  background: radial-gradient(#fcfcfc, #dadada);
  outline: 1px solid white;
}

.poke-image {
  width: 80%;
  height: auto;
  grid-area: cell--large
}
