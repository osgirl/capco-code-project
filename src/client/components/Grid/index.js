import React from 'react';
import Tile from '../Tile';
import URL from '../../variables';
import axios from 'axios';

import './Grid.css';


export default class Grid extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      icons: []
    };
  }

  handleClick = (e) => {
     e.preventDefault();
  }

  componentDidMount() {
    axios(URL)
      .then(resp => resp.data)
      .then((data) => {
        this.setState({ icons: data });
      })
      .catch(err => console.log(err));
  }

  render() {
    return (
      <div className="container" >
        {this.state.icons.map(data => <Tile name={data[0]} src={data[1]} onClick="handleClick" key={data[0]} />)}
      </div>
    );
  }
}
