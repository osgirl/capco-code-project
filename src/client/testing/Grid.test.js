//__tests__/Grid.test.js
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import Grid from '../components/Grid';

describe('<Graph />', () => {
  it('should render a grid', () => {
  // Render a checkbox with label in the document
  const wrapper=shallow(<Grid />);
 expect(wrapper).toMatchSnapshot(); 
  })
})
