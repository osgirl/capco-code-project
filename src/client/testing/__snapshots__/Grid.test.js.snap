// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Graph /> should render a grid 1`] = `
<div
  className="container"
>
  <Rows
    action={[Function]}
    active="amazon"
    tiles={Array []}
  />
  <Rows
    action={[Function]}
    active="npm"
    tiles={Array []}
  />
</div>
`;
